sklearn/linear_model/_glm/glm.py

```diff
@@ -813,33 +815,48 @@ def __init__(
         warm_start=False,
         verbose=0,
     ):
-
         super().__init__(
             alpha=alpha,
             fit_intercept=fit_intercept,
-            family=TweedieDistribution(power=power),
-            link=link,
+            base_loss_class=HalfTweedieLoss,
+            base_loss_params={"power": power},
             max_iter=max_iter,
             tol=tol,
             warm_start=warm_start,
             verbose=verbose,
         )
+        self.link = link
+        self.power = power
 
-    @property
-    def family(self):
```

Comment:
How does a user know it was a private property? This change might need a deprecation cycle.

Reply:
You are right. But this would mean keeping the file `sklearn._loss.glm_distribution.py` because, e.g.,
```python
mod = TweedieRegressor(power=1.5)
mod.family
```
gives `sklearn._loss.glm_distribution.TweedieDistribution`. We should have made it private from the beginning. Note that it only appears in the API doc because it is inherited (from a private parent).

Proposition:
- Let's deprecate `family` and always return a DeprecationWarning.
- Keep current behavor for `PoissonRegressor` and `GammaRegressor` (returning strings  `"poisson"` and `"gamma"`, respectively).
- Breaking change for `TweedieRegressor`, return `"tweedie"` (or `None` or ...)

Reply:
See https://github.com/scikit-learn/scikit-learn/pull/22548/commits/db6c1928a22a1d6c82986b03ab9219e5f1f57e20